### PR TITLE
Convert debit/credit notes into "facturas rectificativas por diferencias"

### PR DESCRIPTION
**Pull request by cavalle**
* Converts debit/credit notes into "facturas rectificativas por diferencias"
* Leaves corrective invoices (i.e. "facturas rectificativas por sustitución") unsupported for the moment
* Works in combination with https://github.com/invopop/gobl/pull/244 for the proper validations

### CODE REVIEW

File: internal/doc/invoice.go [hunk 1]
```diff
@@ -155,6 +178,37 @@ func newClaves(inv *bill.Invoice) []IDClave {
...
+	}
+
+	p := inv.Preceding[0]
+
+	return &FacturasRectificadasSustituidas{
+		IDFacturaRectificadaSustituida: []*IDFacturaRectificadaSustituida{
+			{
+				SerieFactura:           p.Series,
+				NumFactura:             p.Code,
+				FechaExpedicionFactura: formatDate(p.IssueDate),
```
Comment (samlown) [inline internal/doc/invoice.go:206]: I think this is fine. I honestly would not try and extract the original documents date from the signature unless this actually causes an error.

**Review by samlown — APPROVED** [review]
Looks great. 👍  Might be a good idea to document the limitation with the rectified method supported (only credit notes) in the README

File: internal/doc/invoice.go (hunk 1 above)
Comment (cavalle) [reply internal/doc/invoice.go:206]: I checked that no validation is performed at the time of posting. You can actually put any value as the series, num and fecha expedición of the reference invoice. I suppose they do it like this to allow to send a credit note for an invoice that hasn't been posted yet. TBAI seems to have a quite liberal implementation in this regard.

**Comment by cavalle** [comment]
Merging to prepare the PR in the Ticketbai provider

**Pull request merged**